### DISCUSSION TITLE
How to query '%' string?

### DISCUSSION THREAD

**Discussion by tansentcn** [prisma/prisma#23470] [Q&A] [answered]
### Question

```js
  prisma.ws_book_collect_cell
    .findMany({
      where: {
        detail_v: {
          contains: '%' // find all
        }
      }
    })
    .then(res => {
      console.log('res1', res.length) // I want find 1
    })
```

### Information about Prisma Schema, Client Queries and Environment (optional)

@prisma/client: 5.10.2
prisma: 5.10.2
node: v16.13.2
electron: v19.1.9

**Comment by ludralph** [comment]
Hi @tansentcn 👋 

Thank you for raising this question.

if you want to match the literal characters like `%`, you need to escape those characters. For example:
```js
const books = await prisma.ws_book_collect_cell.findMany({
  where: {
    detail_v: {
      contains: '\\%' // note that the `%` character is escaped
    }
  }
})
```
This query will match any book whose `detail_v` contains the literal character `%`. See this section of the [documentation](https://www.prisma.io/docs/orm/prisma-client/queries/filtering-and-sorting#how-does-filtering-work-at-the-database-level)

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by tansentcn** [reply]
Thank you for your reply!
But I think it doesn't work.
![image](https://github.com/prisma/prisma/assets/38307150/a104a970-17c6-407a-8eef-ee11d657367f)

**Reply by tansentcn** [reply]
Environment version:
windows 10
node 16.13.2
npm 8.1.2
Package version:
"prisma": "^5.10.2"
"sqlite3": "^5.1.6"
![image](https://github.com/prisma/prisma/assets/38307150/36cba340-b2a0-4606-8178-ad537ce13433)

**Reply by ludralph** [reply, marked as answer]
The reason it didn't work is because the filtering approach is for MySQL and PostgreSQL as mentioned in the documentation. You may need to write raw queries for SQLite.

**Reply by tansentcn** [reply]
OMG...

Is there a plan to handle sqlite with compatibility?

**Reply by ludralph** [reply]
I searched through our issues and couldn't find one for SQLite compatibility. Can you please create a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) for it?

**Reply by tansentcn** [reply]
OK.
https://github.com/prisma/prisma/issues/23611
